conn_test.go

```diff
@@ -86,14 +86,38 @@ func (p *pipeConn) RemoteAddr() net.Addr               { return nil }
 func (p *pipeConn) SetDeadline(t time.Time) error      { return nil }
 func (p *pipeConn) SetReadDeadline(t time.Time) error  { return nil }
 func (p *pipeConn) SetWriteDeadline(t time.Time) error { return nil }
+func (p *pipeConn) Left() int                          { return p.r.Len() }
```

Comment:
Nit: Suggest renaming to `Len` to avoid confusion with Left/Right, and align with `bytes.Buffer`.

Reply:
Or better yet, since you only ever test `Left() > 0`, just have a method `Empty() bool`